File: year_3/semester_6/PS/tsma/api/src/main/java/pt/isel/tsma/entity/dto/report/ReportDTO.java
package pt.isel.tsma.entity.dto.report;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import pt.isel.tsma.entity.model.shift.Type;

import java.time.LocalDate;

@Data
@NoArgsConstructor
public class ReportDTO implements IReportDTO {

	private long id;

	private long memberId;

	private long calendarId;

	@NonNull
	private LocalDate startDate;

	@NonNull
	private LocalDate endDate;

	@NonNull
	private Type shiftType;

	private int totalShifts;
}

File: year_3/semester_6/PS/tsma/spa/src/app/model/dtos/report/report-dto.ts
import {Type} from "../../type";

export class ReportDTO {

	totalShifts : number
	memberId   : number
	shiftType  : Type

	constructor(totalShifts: number, memberId: number, shiftType: Type) {
		this.totalShifts = totalShifts;
		this.memberId = memberId;
		this.shiftType = shiftType;
	}
}
